Build a regression model.

In [18]:
import pandas as pd
import statsmodels.api as sm

%store -r combined_df
#
partial_df =combined_df.dropna(subset=['Number_of_Bikes', 'Distance'])
#The plan is to create a simple linear regression with distance as independent and Number of free bikes as dependent
X = partial_df[['Distance']]
y = partial_df['Number_of_Bikes']

#Adding a constant
#X = sm.add_constant(X)

# Creating and fitting a linear regression model
regression_model = sm.OLS(y, X)
regression_results = regression_model.fit()

# Print model summary
print(regression_results.summary())



                                 OLS Regression Results                                
Dep. Variable:        Number_of_Bikes   R-squared (uncentered):                   0.623
Model:                            OLS   Adj. R-squared (uncentered):              0.620
Method:                 Least Squares   F-statistic:                              178.6
Date:                Tue, 04 Jul 2023   Prob (F-statistic):                    1.25e-24
Time:                        12:46:57   Log-Likelihood:                         -312.80
No. Observations:                 109   AIC:                                      627.6
Df Residuals:                     108   BIC:                                      630.3
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
                 coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------

Provide model output and an interpretation of the results. 

The R-squared value of 0.623 means that approximately 62.3% of the variance in the Number_of_Bikes can be explained by the independent variable Distance. This suggests that there is a significant relationship between the distance of Point of Interest from the bike stations and the number of free bikes available.The closer the Point of Interest is to the bike stations, the higher the likelihood of having more free bikes. It implies that proximity to restaurants, bars, and groceries plays a role in determining the availability of bikes at the stations.

# Stretch

How can you turn the regression model into a classification model?